pkg/kv/kvserver/rditer/replica_data_iter.go

```diff
@@ -25,6 +25,9 @@ type ReplicaDataIteratorOptions struct {
 	IterKind storage.MVCCIterKind
 	// KeyTypes is passed to underlying iterator to select desired key types.
 	KeyTypes storage.IterKeyType
+	// ExcludeUserKeySpan removes UserKeySpace span portion. This is used if
+	// fast path managed to remove this part of data using ClearRange.
+	ExcludeUserKeySpan bool
```

Comment:
I don't think the bit about fast path/ClearRange belongs here, this iterator is agnostic to who is using it.